fulltext/backends/__hwp.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import absolute_import
+
+import logging
+from html2text import config
+import html2text
+from fulltext.util import run, which
+
+config.BODY_WIDTH = 0
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.addHandler(logging.NullHandler())
+
+EXTENSIONS = ('hwp',)
+
+
+if which('hwp5proc') is None:
+    LOGGER.warning('CLI tool "hwp5proc" is required for .hwp backend. use "pip install pyhwp"')
+
+
+def _cmd(path, **kwargs):
+    cmd = ['hwp5proc', 'xml']
+    cmd.extend([path])
+    return cmd
+
+
+def to_text(html):
+    return html2text.html2text(html)
+
+
+def _get_path(path, **kwargs):
+    return to_text(run(*_cmd(path, **kwargs)).decode('utf-8'))
```

Comment:
I would love to see this backend use the pyhwp library directly (rather than defaulting to the CLI). This may involve less overhead than spawning a separate process.

I understand that the package is not yet compatible with Python 3, so in that case, using the CLI would be the only available option. It may be possible to do a conditional import.

Also, note that there is an HTML backend already. I would like to see this backend use it instead of duplicating that functionality. I was not aware of the `html2text` module, so that might be a better method for the HTML backend's implementation.

Reply:
I  love to conditional importing pyhwp. I'll try someday.